source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+import ChevronLeftIcon from 'octicon/chevron-left.svg';
+import ChevronRightIcon from 'octicon/chevron-right.svg';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import SearchQuery from '../github-helpers/search-query';
+import looseParseInt from '../helpers/loose-parse-int';
+import {getRepoURL, getConversationNumber} from '../github-helpers';
+
+function getButton(direction: string): HTMLAnchorElement {
+	return (
+		<a
+			aria-disabled="true"
+			className="btn btn-lg BtnGroup-item"
+			aria-label={`Navigate to ${direction} Conversation`}
+			data-hotkey={direction.charAt(0)}
+		>
+			{direction === 'next' ? <ChevronRightIcon/> : <ChevronLeftIcon/>}
+		</a>
+	) as unknown as HTMLAnchorElement;
+}
+
+async function init() {
+	const previousButton = getButton('previous');
+	const nextButton = getButton('next');
+
+	select('.gh-header-meta')!.append(
+		<div className="BtnGroup ml-2">
+			{previousButton}
+			{nextButton}
+		</div>
+	);
+
+	const conversationNumber = getConversationNumber();
+	const {list, query, page} = await getConversationList(conversationNumber);
+	const conversation = list.find(item => item.number === conversationNumber);
+
+	setButtonHref(
+		previousButton,
+		query,
+		page,
+		list.find(item => item.cursor === conversation!.cursor - 1)
+	);
+
+	setButtonHref(
+		nextButton,
+		query,
+		page,
+		list.find(item => item.cursor === conversation!.cursor + 1)
+	);
+}
+
+function setButtonHref(button: HTMLAnchorElement, query: string, page: number, conversation?: Conversation) {
+	if (conversation) {
+		const url = new URL(conversation.url);
+		url.searchParams.set('q', query);
+		url.searchParams.set('page', Number(page));
```

Comment:
I think these should not be public. Can you try to move them to `sessionStorage`?

Reply:
Is this still valid?

Reply:
Yes. It changes the feature a bit because the query is either in document.referrer searchParams OR in the sessionsStorage